### PR TITLE
Product Page: Add Tests for Breadcrumbs

### PR DESCRIPTION
**Pull request by batbattur**
## Summary
Added tests that check breadcrumbs on mobile/tablet and desktop.

On mobile and tablet (viewport width sizes below 1000px), we display only the last child breadcrumb
regardless of the number of breadcrumb links. Everything else is collapsed in the breadcrumb menu.
The test for mobile asserts that the last child breadcrumb and the mobile breadcrumb menu are visible.

On the desktop, we display 2 ancestors and the last child breadcrumb (in total 3 breadcrumbs). 
Everything else is collapsed in the breadcrumb menu. The tests for the desktop assert that the 
last child and ancestor breadcrumbs are visible, and the menu is visible/hidden when appropriate.  

## QA notes
Passing CI is enough.

qa_req 0
closes: #1203

### CODE REVIEW

File: frontend/tests/playwright/product/product_breadcrumb.spec.ts [hunk 1]
```diff
@@ -68,6 +68,17 @@ test.describe('Product breadcrumb test', () => {
             ).not.toBeNull();
          }
          expect(ancestorLinksCount).toBeLessThanOrEqual(3);
+
+         // If the product has less than or equal to 3 breadcrumb links, we don't display the breadcrumb menu
+         const desktopBreadcrumbMenu = page.getByTestId(
+            'breadcrumb-menu-desktop'
+         );
+         await expect(desktopBreadcrumbMenu).not.toBeVisible();
```
Comment (ardelato) [inline frontend/tests/playwright/product/product_breadcrumb.spec.ts:76]: If we are not using the set initialized variable later in the code why not pass the retrieval of the element directly to the assertion?

```suggestion
         await expect(page.getByTestId(
            'breadcrumb-menu-desktop'
         )).not.toBeVisible();
```

**Review by ardelato — APPROVED** [review]
CR 👍 with minor suggestion

File: frontend/tests/playwright/product/product_breadcrumb.spec.ts (hunk 1 above)
Comment (batbattur) [reply frontend/tests/playwright/product/product_breadcrumb.spec.ts]: Thanks @ardelato for the suggestion! That makes it very cleaner! Updated to use the elements directly for the unused variables in https://github.com/iFixit/react-commerce/pull/1227/commits/7e5c2b4f96ac281a72e3117878516acd67cc5d31.

**Review by mlahargou — APPROVED** [review]
CR 📱 Tests look good!

**Pull request merged**